src/ralph/virtual/subscriptions.py

```diff
@@ -0,0 +1,49 @@
+import logging
+
+import pkg_resources
+import pyhermes
+from django.conf import settings
+from django.utils.text import slugify
+
+
+
+logger = logging.getLogger(__name__)
+
+
+ENTRY_POINTS_GROUP = 'ralph.dchost_processors'
+ENDPOINT_PREFIX = 'cloud-sync-'
+DCHOST_PROCESSORS = {}
+SUBSCRIPTIONS = {}
+
+
+def load_processors():
+    for ep in pkg_resources.iter_entry_points(ENTRY_POINTS_GROUP):
+        try:
+            DCHOST_PROCESSORS[ep.name] = ep.resolve()
+        except ImportError:
+            logger.error(
+                'Could not import DC asset event processor from {}.'
+                ''.format(ep.module_name)
+            )
+
+
+def generate_listeners():
+    load_processors()
+
+    from ralph.virtual.models import CloudProvider
+    cloud_providers = CloudProvider.objects.filter(
```

Comment:
what if I add new cloud provider? why I'll have to restart ralph to create new subscriber?

Reply:
For now yes -- it's necessary to restart Ralph in order to regenerate the list of listeners. The plan is to regenerate them on post_save signal.

Reply:
Right, for now it will be necessary to restart Ralph. I'm going to add a post_save signal subscriber in order to regenerate the list of subscribers every time something is done to a cloud provider. However, I'd rather did it in a separate patch.

Reply:
why couldn't you make one generic endpoint? or pass cloudprovider info as query param?